### PR TITLE
Add organization endpoints

### PR DESCRIPTION
**Pull request by eloaverona**
This PR implements the endpoints GET /organization and GET/organization/{id}.  It includes tests and updates to the REST API spec doc. 

To run the tests:
`cargo test --features test-api --  --test-threads=1 `

~Based on #56.~

### CODE REVIEW

File: daemon/src/rest_api/routes/mod.rs [hunk 1]
```diff
@@ -473,6 +492,161 @@ mod test {
...
+    ///
+    /// Verifies a GET /organization responds with an Ok response
+    ///     with a list containing one organization
+    ///
+    #[test]
+    fn test_list_organizations() {
+        database::run_migrations(&DATABASE_URL).unwrap();
+        let test_pool = get_connection_pool();
+        let mut srv = create_test_server(ResponseType::ClientBatchStatusResponseOK);
+        // Clears the agents table in the test database
```
Comment (shannynalayna) [inline daemon/src/rest_api/routes/mod.rs]: This comment isn't relevant/ should be `organization`

File: daemon/src/rest_api/routes/mod.rs [hunk 2]
```diff
@@ -473,6 +492,161 @@ mod test {
...
+    ///
+    /// Verifies a GET /organization/{id} responds with NotFound response
+    /// when there is no organization with the specified id.
+    ///
+    #[test]
+    fn test_fetch_organization_not_found() {
+        database::run_migrations(&DATABASE_URL).unwrap();
+        let test_pool = get_connection_pool();
+        let mut srv = create_test_server(ResponseType::ClientBatchStatusResponseOK);
+        // Clears the agents table in the test database
```
Comment (shannynalayna) [inline daemon/src/rest_api/routes/mod.rs]: The same comment copied here, should be `organizations table`

File: daemon/openapi.yaml [hunk 3]
```diff
@@ -209,6 +209,71 @@ paths:
...
+        500:
+          description: Something went wrong within the database
+          schema:
+            $ref: "#/definitions/Error"
+        503:
+          description: API is unable to reach the database
+          schema:
+            $ref: "#/definitions/Error"
+
+
```
Comment (dplumb94) [inline daemon/openapi.yaml]: Extra newline

File: daemon/src/rest_api/routes/mod.rs [hunk 4]
```diff
@@ -565,6 +568,85 @@ mod test {
...
+        let response = srv.execute(request.send()).unwrap();
+        assert!(response.status().is_success());
+        let org: OrganizationSlice =
+            serde_json::from_slice(&*response.body().wait().unwrap()).unwrap();
+        assert_eq!(org.name, ORG_NAME_1.to_string());
+        assert_eq!(org.org_id, KEY2.to_string());
+        assert_eq!(org.address, ADDRESS_1.to_string());
+    }
+
+    ///
```
Comment (dplumb94) [inline daemon/src/rest_api/routes/mod.rs]: Extra comment line

File: daemon/src/rest_api/routes/mod.rs (hunk 2 above)
Comment (eloaverona) [reply daemon/src/rest_api/routes/mod.rs]: ooops... fixed! thanks for catching that!

File: daemon/src/rest_api/routes/mod.rs [hunk 5]
```diff
@@ -473,6 +489,83 @@ mod test {
         assert_eq!(agent.org_id, KEY2.to_string());
     }
 
+    ///
+    /// Verifies a GET /organization responds with an Ok response
```
Comment (dplumb94) [inline daemon/src/rest_api/routes/mod.rs]: Does this test cover both of these comments? I think it should just be two tests.

File: daemon/src/rest_api/routes/mod.rs [hunk 6]
```diff
@@ -565,7 +568,85 @@ mod test {
...
+        let org: OrganizationSlice =
+            serde_json::from_slice(&*response.body().wait().unwrap()).unwrap();
+        assert_eq!(org.name, ORG_NAME_1.to_string());
+        assert_eq!(org.org_id, KEY2.to_string());
+        assert_eq!(org.address, ADDRESS_1.to_string());
+    }
+
+    ///
+    ///
+    /// Verifies a GET /organization/{id} responds with an Ok response
```
Comment (dplumb94) [inline daemon/src/rest_api/routes/mod.rs]: This test is valid with the current implementation, but we definitely need to update these routes to have a "Head Block" param. Clients may want to coordinate data at a certain block height, especially if we're using a forking consensus.

Comment (eloaverona) [reply daemon/src/rest_api/routes/mod.rs]: It's a good point, but I think this should be a separate story, to fix across all the routes already implemented.

File: daemon/src/rest_api/routes/mod.rs [hunk 7]
```diff
@@ -554,4 +734,46 @@ mod test {
...
+                name: ORG_NAME_2.to_string(),
+                address: UPDATED_ADDRESS_2.to_string(),
+                metadata: vec![],
+                start_block_num: 4,
+                end_block_num: database::helpers::MAX_BLOCK_NUM,
+            },
+        ]
+    }
+
+    fn populate_organiation_table(conn: &PgConnection, organizations: Vec<NewOrganization>) {
```
Comment (shannynalayna) [inline daemon/src/rest_api/routes/mod.rs]: Noticed another small typo here: `organiation`

Comment (eloaverona) [reply daemon/src/rest_api/routes/mod.rs]: Addressed!

**Pull request merged**